pennylane/ops/op_math/sum.py

```diff
@@ -179,7 +179,7 @@ def is_hermitian(self):
         return (
             all(s.is_hermitian for s in self)
             if self._pauli_rep is None
-            else not any(qml.math.iscomplex(val) for val in self._pauli_rep.values())
+            else not math.any(math.iscomplex(self._pauli_rep.values()))
```

Comment:
a `tf.Tensor` being trained can't be iterated over, so this failed. need to use `math.any` because `math.iscomplex` results depend on the interface:
```pycon
>>> math.iscomplex(np.array([0.1, 0.2j]))
array([False,  True])
>>> math.iscomplex(tf.convert_to_tensor([0.1j, 0.2j]))
<tf.Tensor: shape=(), dtype=bool, numpy=True> # this would break regular "any" because it's not iterable
```